src/unix/darwin.c

```diff
@@ -333,3 +333,29 @@ void uv_free_interface_addresses(uv_interface_address_t* addresses,
 
   uv__free(addresses);
 }
+
+int uv__tcp_enable_largesocket(uv_tcp_t *tcp) {
+  int rmem;
+  int wmem;
+  int r;
+
+  size_t size = sizeof(int);
+  if (sysctlbyname("net.inet.tcp.recvspace", &rmem, &size, NULL, 0) ||
+      rmem <= STDTCPWINDOW)
+    return UV_ENOSYS;
+  
+  if (sysctlbyname("net.inet.tcp.sendspace", &wmem, &size, NULL, 0) ||
```

Comment:
@santigimeno - the darwin specific code is now ported onto BSD. Few points though:
* The CI bot is freebasd only, so I wasn't able to test for other BSD variants. But looking at the documentation for the sysctl API as well as another documentation which compares all the BSD variants, I see that the code applies to at least 5 platforms which derive their base system capabilities from 4.4BSD: freebsd, netbsd, openbsd, macos, ios.

* There is a mention about dragonfly BSD variant in the documentation, and usage of it in libuv code. I wasn't sure about its relevance in this context, so omitted it for the time. What do you suggest?